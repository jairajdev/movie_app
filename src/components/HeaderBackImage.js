import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';

const HeaderBackImage = () => {
  return <Icon size={25} color={'white'} name="chevron-back" />;
};
export default HeaderBackImage;
