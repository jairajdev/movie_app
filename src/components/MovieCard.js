import React from 'react';
import {Image, StyleSheet, TouchableOpacity, Text, View} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

const IMAGE_PATH = 'https://image.tmdb.org/t/p/w500';

const MovieCard = (props) => {
  const {movie, navigation} = props;
  const {img, starStyle, textStyle, box} = styles;

  const imageUrl = IMAGE_PATH + movie.poster_path;
  const title = movie.title;
  const vote_average = movie.vote_average;

  const render_rating_star = () => {
    const rating_star = [];

    for (let i = 0; i < vote_average / 2; i++) {
      rating_star.push(
        <FontAwesome
          key={i}
          name="star"
          size={18}
          color="#F09839"
          style={starStyle}
        />,
      );
    }

    if (rating_star.length < 5) {
      rating_star.push(
        <FontAwesome
          key={98}
          name="star-o"
          size={18}
          color="black"
          style={starStyle}
        />,
      );
    }

    return rating_star;
  };

  return (
    <TouchableOpacity
      onPress={() => navigation.navigate('MovieDetail', {movieId: movie.id})}>
      <Image style={img} source={{uri: imageUrl}} />
      <Text style={textStyle}>{title}</Text>
      <View style={box}>{render_rating_star(vote_average)}</View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  img: {
    height: 120,
    borderRadius: 5,
    resizeMode: 'cover',
    marginBottom: 5,
  },
  textStyle: {
    fontSize: 15,
  },
  box: {
    flexDirection: 'row',
    marginVertical: 10,
  },
  starStyle: {
    marginHorizontal: 1,
  },
});

export default MovieCard;
