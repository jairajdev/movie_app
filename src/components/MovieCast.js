import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

const MovieCast = (props) => {
  const {state} = props;

  const {container, box, boldText} = styles;
  const {charachters, director, production} = state;

  return (
    <View style={container}>
      {charachters.map((name, key) => {
        return (
          <View style={box} key={key}>
            <Text style={boldText}>{name}</Text>
            <Text>Cast</Text>
          </View>
        );
      })}

      <View style={box}>
        <Text style={boldText}>{director}</Text>
        <Text>Director</Text>
      </View>

      {production.map((name, key) => {
        return (
          <View style={box} key={key}>
            <Text style={boldText}>{name}</Text>
            <Text>Production</Text>
          </View>
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    borderColor: 'red',
    color: 'red',
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  box: {
    width: 100,
    flexGrow: 1,
    fontSize: 55,
    padding: 20,
  },
  boldText: {
    paddingBottom: 3,
    fontWeight: 'bold',
  },
});

export default MovieCast;
