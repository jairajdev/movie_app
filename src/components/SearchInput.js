import React, {useState} from 'react';
import {SearchBar} from 'react-native-elements';

import useSearchMovie from '../utils/useSearchMovie';

const SearchInput = (props) => {
  const [inputState, setState] = useState('');

  const {searchOn, searchOff, handleSearchQuery} = props;

  const {clearSearchMovies} = useSearchMovie();

  const handle_input = (value) => {
    setState(value);
    searchOn();
    handleSearchQuery(value);
  };

  const handle_clear = () => {
    clearSearchMovies();
    setState('');
  };

  return (
    <SearchBar
      onTouchStart={searchOn}
      onChangeText={(text) => handle_input(text)}
      onClear={handle_clear}
      onCancel={searchOff}
      value={inputState}
      platform="ios"
      placeholder="Search"
    />
  );
};

export default SearchInput;
