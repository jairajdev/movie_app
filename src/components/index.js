export {default as SearchInput} from './SearchInput';
export {default as MovieCast} from './MovieCast';
export {default as MovieCard} from './MovieCard';
export {default as MovieList} from './MovieList';
export {default as HeaderBackImage} from './HeaderBackImage';
