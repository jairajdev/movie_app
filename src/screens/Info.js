import React from 'react';
import {StyleSheet, Text, View, Image} from 'react-native';

const Info = () => {
  const {
    container,
    imageStyle,
    yellowBox,
    overviewTitle,
    overviewStyle,
    box,
    yellowText,
  } = styles;

  return (
    <View style={container}>
      <Image source={require('../image/tmdb.png')} style={imageStyle} />
      <View style={box}>
        <Text style={overviewTitle}>Welcome.</Text>
        <Text style={overviewStyle}>
          Millions of movies, TV shows and people to discover. Explore now.
        </Text>
      </View>
      <View style={yellowBox}>
        <Text style={yellowText}>
          The data for this app are fetched from the API of The Movie DataBase.
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  imageStyle: {
    height: 200,
    width: 200,
  },
  overviewTitle: {
    color: '#0B253F',
    fontSize: 25,
    fontWeight: '600',
  },
  overviewStyle: {
    fontSize: 20,
    color: '#222',
  },
  box: {
    flexDirection: 'column',
    alignItems: 'center',
    padding: 20,
  },
  yellowBox: {
    alignItems: 'center',
    padding: 20,
    backgroundColor: '#f09839',
    borderRadius: 5,
    margin: 10,
  },
  yellowText: {
    fontSize: 18,
    color: '#FFF',
  },
});

export default Info;
