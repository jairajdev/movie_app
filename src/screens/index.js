export {default as Movie} from './Movie';
export {default as MovieDetail} from './MovieDetail';
export {default as Info} from './Info';
