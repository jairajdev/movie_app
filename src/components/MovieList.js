import React from 'react';
import {FlatList, View, StyleSheet} from 'react-native';

import MovieCard from './MovieCard';

const MovieList = (props) => {
  const {movieList, navigation} = props;
  const {card, container} = styles;

  return (
    <View style={container}>
      <FlatList
        numColumns={2}
        data={movieList}
        keyExtractor={(item) => {
          return item.id;
        }}
        renderItem={(movie) => {
          return (
            <View style={card}>
              <MovieCard movie={movie.item} navigation={navigation} />
            </View>
          );
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  card: {
    flex: 1,
    flexDirection: 'column',
    margin: 10,
    padding: 10,
    backgroundColor: '#FFFFFF',
    elevation: 1,
    borderRadius: 5,
  },
});
export default MovieList;
