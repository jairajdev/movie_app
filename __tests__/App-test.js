import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock.js';

import {Movie, Info} from '../src/screens';
import {SearchInput} from '../src/components';
import {
  getData,
  getSearchMovieUrl,
  getMovieDetailUrl,
  getMovieCreditUrl,
} from '../src/utils/api';

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo);

test('movie', () => {
  const tree = renderer.create(<Movie />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('info', () => {
  const tree = renderer.create(<Info />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('search input', () => {
  const tree = renderer.create(<SearchInput />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('search movie api test', async () => {
  const response = await getData(getSearchMovieUrl('Mulan', 1));

  expect(response).toBeDefined();
  expect(response.results).toBeDefined();

  if (response) {
    response.results.map((item) => {
      expect(item.id).toBeDefined();
      expect(typeof item.id).toBe('number');

      expect(item.title).toBeDefined();
      expect(typeof item.title).toBe('string');

      expect(item.vote_average).toBeDefined();
      expect(typeof item.vote_average).toBe('number');
    });
  }
});

test('movie detail api test', async () => {
  const mulan_movie_id = '337401';

  const responseDetail = await getData(getMovieDetailUrl(mulan_movie_id));
  const responseCrew = await getData(getMovieCreditUrl(mulan_movie_id));

  expect(responseDetail).toBeDefined();
  expect(responseCrew).toBeDefined();

  if (responseDetail) {
    expect(responseDetail.title).toBeDefined();
    expect(typeof responseDetail.title).toBe('string');

    expect(responseDetail.vote_average).toBeDefined();
    expect(typeof responseDetail.vote_average).toBe('number');

    expect(responseDetail.overview).toBeDefined();
    expect(typeof responseDetail.overview).toBe('string');

    expect(responseDetail.popularity).toBeDefined();
    expect(typeof responseDetail.popularity).toBe('number');

    expect(responseDetail.release_date).toBeDefined();
    expect(typeof responseDetail.release_date).toBe('string');

    expect(responseDetail.vote_count).toBeDefined();
    expect(typeof responseDetail.vote_count).toBe('number');
  }
});
