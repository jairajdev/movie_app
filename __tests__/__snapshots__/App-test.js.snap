// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/image/tmdb.png",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0B253F",
          "fontSize": 25,
          "fontWeight": "600",
        }
      }
    >
      Welcome.
    </Text>
    <Text
      style={
        Object {
          "color": "#222",
          "fontSize": 20,
        }
      }
    >
      Millions of movies, TV shows and people to discover. Explore now.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f09839",
        "borderRadius": 5,
        "margin": 10,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    >
      The data for this app are fetched from the API of The Movie DataBase.
    </Text>
  </View>
</View>
`;

exports[`movie 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "overflow": "hidden",
        "paddingBottom": 13,
        "paddingTop": 13,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#dcdce1",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 9,
            "flexDirection": "row",
            "marginLeft": 8,
            "marginRight": 8,
            "minHeight": 36,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#7d7d7d",
                        "fontSize": 20,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onBlur={[Function]}
          onCancel={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onTouchStart={[Function]}
          placeholder="Search"
          placeholderTextColor="#7d7d7d"
          platform="ios"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 6,
              "minHeight": 40,
              "overflow": "hidden",
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginRight": 8,
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "height": 0,
            "margin": 0,
            "padding": 0,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "opacity": 0,
          "position": "absolute",
          "right": -0,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <Text
    style={
      Object {
        "color": "#0B253F",
        "fontSize": 20,
        "margin": 15,
      }
    }
  >
    Search by movie title
  </Text>,
]
`;

exports[`search input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "overflow": "hidden",
      "paddingBottom": 13,
      "paddingTop": 13,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#dcdce1",
          "borderBottomWidth": 0,
          "borderColor": "#86939e",
          "borderRadius": 9,
          "flexDirection": "row",
          "marginLeft": 8,
          "marginRight": 8,
          "minHeight": 36,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#7d7d7d",
                      "fontSize": 20,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onCancel={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Search"
        placeholderTextColor="#7d7d7d"
        platform="ios"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "marginLeft": 6,
            "minHeight": 40,
            "overflow": "hidden",
          }
        }
        testID="searchInput"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginRight": 8,
            "marginVertical": 4,
            "paddingRight": 4,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "height": 0,
          "margin": 0,
          "padding": 0,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "right": -0,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
